src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -1,537 +1,61 @@
 package codeu.chat.client.simplegui;
 
-import codeu.chat.client.ClientUser;
-import java.io.File;
 import javafx.application.Application;
-import javafx.application.Platform;
-import javafx.event.ActionEvent;
-import javafx.geometry.Pos;
-import javafx.scene.Scene;
-import javafx.scene.control.*;
-import javafx.scene.layout.*;
-import javafx.scene.paint.Color;
-import javafx.scene.paint.Paint;
 import javafx.stage.Stage;
-import javafx.scene.text.*;
-import javafx.collections.*;
-import javafx.scene.input.MouseEvent;
-import javafx.scene.control.Alert.AlertType;
-
-import java.util.Map;
 
 import codeu.chat.client.ClientContext;
 import codeu.chat.client.Controller;
-import codeu.chat.common.ConversationSummary;
-import codeu.chat.common.User;
-import codeu.chat.common.Message;
 import codeu.chat.client.View;
 import codeu.chat.util.Logger;
-import codeu.chat.util.Uuid;
-import codeu.chat.util.store.Store;
 
 public final class ChatGuiFX extends Application {
 
     private final static Logger.Log LOG = Logger.newLog(ChatGuiFX.class);
 
-    private static final double WINDOW_WIDTH = 1000;
-    private static final double WINDOW_HEIGHT = 500;
-    private static final String SIGNIN_ERROR_MESSAGE = "Your username or password does not match. Have you signed up yet?";
-    private static final String SIGNUP_ERROR_MESSAGE = "Sorry, that username already exists. Please choose a different one.";
-    private static final String BADCHAR_ERROR_MESSAGE = "Usernames and passwords can only be composed of letters and numbers, no special characters.";
-
-    // both page vars
-    //private ClientContext clientContext;
-
-    /*
-        **This is something to think about**
-        // TODO: split the login & main screens into separate methods
-        public void run(String[] args) {
-            buildLoginScene();
-            buildMainScene();
-        }
-    */
+    protected static final double WINDOW_WIDTH = 1000;
+    protected static final double WINDOW_HEIGHT = 500;
 
-    // login page vars
     // Holds the scene that user is currently viewing
     private Stage thestage;
 
-    // Scenes to hold all the elements for each page
-    private Scene signInScene, mainScene;
-
-    // Takes input for username and password for sign in/up
-    private TextField userInput;
-    private PasswordField passInput;
-
-    // Displays error messages for when the user is not sign in or has not selected a convo
-    private Label errorLabel;
+    private MainChatPage mainPage;
```

Comment:
I think so, yeah, because I want to use an instance of the main page in methods other than start() (specifically, switchToMainPage()). If you have an alternative solution lmk though!